cloudinit/distros/__init__.py

```diff
@@ -720,6 +725,104 @@ def create_group(self, name, members=None):
                 LOG.info("Added user '%s' to group '%s'", member, name)
 
 
+def _apply_hostname_transformations_to_url(
```

Comment:
What do you think about adding this to url_helper.py instead?

Reply:
As things stand, this function does not have any standalone unit tests, because it's tested as part of the larger unit. Moving it to `url_helper` would (to me, at least) imply an expectation that it can be used more generally, which would in turn imply that it should have its own testing.

I don't object to it being in `url_helper.py`, but I wonder if (a) we can do that as a follow-up, and (b) if we should instead defer such a move, and the work involved, until we have a potential second caller.

Reply:
OK.  I think until we push out urllib from cloudinit.util  we won't pay any more of a cost by importing net related modules where we don't really need them.